pkg/cloud/gcp/actuators/machine/machine_scope.go

```diff
@@ -85,12 +89,33 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 		}
 	}
 
-	computeService, err := params.computeClientBuilder(serviceAccountJSON)
+	var computeEndpoint *configv1.GCPServiceEndpoint = nil
+	var tagsEndpoint *configv1.GCPServiceEndpoint = nil
+	if sets.New(params.featureGates.KnownFeatures()...).Has(string(apifeatures.FeatureGateGCPCustomAPIEndpoints)) {
```

Comment:
You don't need to check if the feature is known, if the feature is not known, then something has gone very wrong